keyring/keychain.go

```diff
@@ -323,6 +286,27 @@ func (k *keychain) Keys() ([]string, error) {
 // -------------------------------------------------
 // OSX Keychain API funcs
 
+func keychainExists(path string) (bool, error) {
+	// returns no error even if it doesn't exist
+	k, err := openKeychain(path)
+	if err != nil {
+		log.Printf("Error opening %v", err)
+		return false, err
+	}
+	defer C.CFRelease(C.CFTypeRef(k))
+
+	var status C.SecKeychainStatus
```

Comment:
I'm pretty sure that SecKeychainStatus doesn't need to be explicitly released here, it's not a ref.